docker/docker_image_dest.go

```diff
@@ -298,18 +309,43 @@ func (d *dockerImageDestination) TryReusingBlob(ctx context.Context, info types.
 		return false, types.BlobInfo{}, err
 	}
 	if exists {
-		cache.RecordKnownLocation(d.ref.Transport(), bicTransportScope(d.ref), info.Digest, newBICLocationReference(d.ref))
-		return true, types.BlobInfo{Digest: info.Digest, Size: size}, nil
+		var compressorName string
+		if info.CompressionAlgorithm != nil {
+			compressorName = info.CompressionAlgorithm.Name()
+		}
```

Comment:
I'm not completely sure we end up recording the right value here.

Reply:
Changed the `TryReusingBlob()` case where the registry indicates that it already has the blob so that we record the compression type as UnknownCompression.  The MIME type we're passed from `copy` comes from the source image's manifest, so it should already be correct for the blob with the same digest without needing any fixup from us.